distributed_index_tests/lib/user.py

```diff
@@ -1,11 +1,188 @@
-import base64
+import requests
+import concurrent.futures
 import json
+import pprint
+import copy_reg
+import types
+import time
+import base64
+import document
+import uuid
+import re
+from collections import defaultdict
+
+# This function is added to use ProcessExecutor
+# concurrent.futures. 
+# 
+def _pickle_method(m):
+    if m.im_self is None:
+        return getattr, (m.im_class, m.im_func.func_name)
+    else:
+        return getattr, (m.im_self, m.im_func.func_name)
 
-class User:
 
-    def __init__(self, name, password, channels):
+copy_reg.pickle(types.MethodType, _pickle_method)
+
+
+class User:
+    def __init__(self,target_url,db,name,password,channels,uuid=False):
+        self.target_url = target_url
```

Comment:
I think we should construct the base endpoint to use in all requests, similar to this. This will be more readable and can be reused throughout the class and will simplify our endpoint queries. https://github.com/couchbaselabs/sync-gateway-tests/blob/master/distributed_index_tests/lib/syncgateway.py#L45-L46